### ISSUE TITLE
Use a proper Url data type for the `OktaAuth0` / `OktaConfig` structs

### ISSUE THREAD

**Issue by etorreborre** [build-trust/ockam#5346]
[The `OktaAuth0` struct](https://github.com/build-trust/ockam/blob/ce251fb89d4542d03eddcc1d0db25e04072964c3/implementations/rust/ockam/ockam_api/src/cloud/project.rs#L153) and [`OktkaConfig` struct](https://github.com/build-trust/ockam/blob/ce251fb89d4542d03eddcc1d0db25e04072964c3/implementations/rust/ockam/ockam_api/src/cloud/project.rs#L105) currently accept a `String` for the `tenant_base_url`:
```rust
pub struct OktaAuth0 {
    pub tenant_base_url: String,
    pub client_id: String,
    pub certificate: String,
}
```

We should use [a `Url`](https://crates.io/crates/url) instead.

**Comment by aleksandarskrbic** [comment]
I would like to take this one! @etorreborre

**Comment by etorreborre** [comment]
Please do @aleksandarskrbic!

**Comment by etorreborre** [comment]
Fixed with #5368
